infor-design/enterprise
# 3029/3704 - Improve Personalization of Buttons/Vertical Tabs/App Menu

**Explain the _details_ for making this change. What existing problem does the pull request solve?**
This PR makes a number of improvements to the IDS Personalization system combined with some components:
- The personalized Application Menu now looks closer to its non-personalized counterpart in the Soho theme (#3704)
- Vertical Tabs responsive view (basically Header Tabs) can now be personalized, and had some errors in color resolved. (#3029)
- Buttons, hyperlinks, and some standalone typography now take on various shades of the base color when a page section is personalized.
- Added a color utility function for desaturating colors, along with a sandbox test page for the color utilities overall.

**Related github/jira issue (required)**:
Closes #3029 
Closes #3704

**Steps necessary to review your pull request (required)**:
Pull this branch, build, and run the demoapp.

_Test the Application Menu changes:_
- Open http://localhost:4000/components/applicationmenu/example-personalized-roles.html and http://localhost:4000/components/applicationmenu/example-personalized-roles.html?theme=soho&variant=light&colors=db7726
- Ensure that on the Soho theme, the application menu doesn't completely take on the base color.  Only the focus states, borders, and button ripple effects should be personalized.
- Check Uplift (Which SHOULD take on the base color and various shades of it, and should be unchanged): http://localhost:4000/components/applicationmenu/example-personalized-roles.html?theme=uplift&variant=light&colors=db7726

_Test Buttons:_
- Open http://localhost:4000/components/button/example-index.html and http://localhost:4000/components/button/example-index.html?colors=2578a9
- Compare the custom colors on all theme variants to the "defaults".  When comparing "Azure" to "Default" the colors should be relatively close (they may not be 100% due to being calculated).
- Test Uplift: http://localhost:4000/components/button/example-index.html?theme=uplift&variant=light and http://localhost:4000/components/button/example-index.html?colors=0066D4&theme=uplift&variant=light
- Do a general smoke test of buttons on the various color schemes.

_Test Vertical Tabs:_
- http://localhost:4000/components/tabs-vertical/example-responsive.html and http://localhost:4000/components/tabs-vertical/example-responsive.html?theme=soho&variant=light&colors=1F9254
- Test the full page view
- Shrink the browser viewport so the tab container transforms to standard header tabs.  Ensure the scrollable tab area is personalized and take on the correct color in the "faded" parts of the edges.
- Also test the same items on Uplift: http://localhost:4000/components/tabs-vertical/example-responsive.html?theme=uplift&variant=light&colors=1F9254

_Color Utilities Test Page:_
- http://localhost:4000/utils/example-color-saturation.html
- No specific checks for this, just for grasping what the utilities are doing under the hood with some UI controls.

**Included in this Pull Request**:
